internal/gensupport/send.go

```diff
@@ -50,7 +52,7 @@ func send(ctx context.Context, client *http.Client, req *http.Request) (*http.Re
 // If ctx is non-nil, it calls all hooks, then sends the request with
 // req.WithContext, then calls any functions returned by the hooks in
 // reverse order.
-func SendRequestWithRetry(ctx context.Context, client *http.Client, req *http.Request) (*http.Response, error) {
+func SendRequestWithRetry(ctx context.Context, client *http.Client, req *http.Request, retry *RetryConfig) (*http.Response, error) {
```

Comment:
The only thing I think will be an issue is that the remote offset is not considered in the case Storage has progressed and the client is now unaligned.

Reply:
Yeah, that work will have to come in a separate PR. This PR doesn't change the mechanics of the resumption strategy.